### DISCUSSION TITLE
Does using glob or _exists read a file and put in the s3fs cache?

### DISCUSSION THREAD

**Discussion by asant-leitha** [dask/s3fs#926] [Q&A]
If I use first glob or _exists, does put the file in the cache for later usage or not?

**Comment by martindurant** [comment]
The file details (not the file itself) are generally stored after real listing operations on the remote. They might *not* be stored when requesting partial listings, and of course the cache itself is configurable. 

You can turn on logging (e.g., `fsspec.utils.setup_logging(logger_name="s3fs")`), or check the contents of your filesystems `.dircache` attribute to get more specifics on what happens when.
